### PR TITLE
fix: make `handleCreateBucketSynPackage` forward compatible

### PR DESCRIPTION
**Pull request by Pythonberg1997**
### Description

This pr is to fix the forward incompatibility of `handleCreateBucketSynPackage`

### Rationale

To make the chain forward compatible

### Example

The chain can handle `CreateBucketSynPackage` with or without `GlobalVirtualGroupFamilyId`

### Changes

Notable changes: 
* add `handleCreateBucketSynPackageV2`

### CODE REVIEW

File: x/storage/keeper/cross_app_bucket.go [hunk 1]
```diff
@@ -204,7 +224,24 @@ func (app *BucketApp) handleCreateBucketFailAckPackage(ctx sdk.Context, appCtx *
...
 
+func (app *BucketApp) handleCreateBucketFailAckPackageV2(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, synPackage *types.CreateBucketSynPackageV2) sdk.ExecuteResult {
+	app.storageKeeper.Logger(ctx).Error("received create bucket fail ack package ")
+
+	return sdk.ExecuteResult{}
+}
+
 func (app *BucketApp) handleCreateBucketSynPackage(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, createBucketPackage *types.CreateBucketSynPackage) sdk.ExecuteResult {
+	if ctx.IsUpgraded(upgradetypes.Pampas) {
+		return sdk.ExecuteResult{
```
Comment (forcodedancing) [inline x/storage/keeper/cross_app_bucket.go:235]: Do we need `err := createBucketPackage.ValidateBasic()` ?

**Review by zjubfd — APPROVED** [review]

**Comment by zjubfd** [comment]
Looks good to me, please ensure it is well tested locally.

File: x/storage/keeper/cross_app_bucket.go (hunk 1 above)
Comment (Pythonberg1997) [reply x/storage/keeper/cross_app_bucket.go:235]: It's on the below

File: x/storage/types/crosschain.go [hunk 2]
```diff
@@ -354,6 +440,18 @@ type CreateBucketSynPackage struct {
...
+	Visibility                     uint32
+	PaymentAddress                 common.Address
+	PrimarySpAddress               common.Address
+	PrimarySpApprovalExpiredHeight uint64
+	PrimarySpApprovalSignature     []byte
+	ChargedReadQuota               uint64
+	ExtraData                      []byte
+}
+
+type CreateBucketSynPackageV2Struct struct {
```
Comment (forcodedancing) [inline x/storage/types/crosschain.go:454]: What's the difference between `CreateBucketSynPackageV2Struct` and `CreateBucketSynPackageV2`? Why we need both of them?

Comment (Pythonberg1997) [reply x/storage/types/crosschain.go:454]: `CreateBucketSynPackageV2Struct` has an extra field `GlobalVirtualGroupFamilyId`

Comment (forcodedancing) [reply x/storage/types/crosschain.go:454]: I mean `V2Struct` and `V2`.

**Review by yutianwu — APPROVED** [review]

File: x/storage/types/crosschain.go [hunk 3]
```diff
@@ -123,6 +120,7 @@ var (
 		},
 	}
 
+	// DeserializeFuncMapV2 used after Pamela upgrade
```
Comment (forcodedancing) [inline x/storage/types/crosschain.go:123]: Wrong comment `Pamela`

**Review by forcodedancing — APPROVED** [review]

**Pull request merged**